typescript/sdk/src/deploy/middleware/TokenBridgeApp.ts

```diff
@@ -1,7 +1,161 @@
+import fetch from 'cross-fetch';
+import { ethers } from 'ethers';
+
+import {
+  CircleBridgeAdapter__factory,
+  ICircleBridge__factory,
+  ICircleMessageTransmitter__factory,
+} from '@hyperlane-xyz/core';
+
 import { HyperlaneApp } from '../../HyperlaneApp';
+import { Chains } from '../../consts/chains';
+import { MultiProvider } from '../../providers/MultiProvider';
 import { TokenBridgeContracts } from '../../middleware';
-import { ChainName } from '../../types';
+import { ChainMap, ChainName } from '../../types';
+import { objMap } from '../../utils/objects';
+
+import {
+  BridgeAdapterConfig,
+  BridgeAdapterType,
+  CircleBridgeAdapterConfig,
+} from './TokenBridgeRouterDeployer';
+
+const CircleBridgeInterface = ICircleBridge__factory.createInterface();
+const CircleBridgeAdapterInterface =
+  CircleBridgeAdapter__factory.createInterface();
 
+interface CircleBridgeMessage<Chain> {
+  chain: Chain;
+  remoteChain: Chain;
+  txHash: string;
+  message: string;
+  nonce: number;
+  domain: number;
+  nonceHash: string;
+}
```

Comment:
these should ultimately live elsewhere in the SDK

Reply:
Agreed Eventually